src/algebra/ordered_ring.lean

```diff
@@ -148,6 +148,28 @@ lemma one_le_mul_of_one_le_of_one_le {a b : α} (a1 : 1 ≤ a) (b1 : 1 ≤ b) :
   (1 : α) ≤ a * b :=
 (mul_one (1 : α)).symm.le.trans (mul_le_mul a1 b1 zero_le_one (zero_le_one.trans a1))
 
+/-- Pullback an `ordered_semiring` under an injective map. -/
+def function.injective.ordered_semiring {β : Type*}
```

Comment:
Are you willing to add this for `ordered_ring` within the same file? Or do you want to leave that for a future PR?

Reply:
`ordered_comm_semiring` should be trivial at least.

Reply:
Sure, I can do it, but it will be tomorrow!

Reply:
At the moment there are 
* `function.injective.ordered_semiring`, with the `to_ordered_semiring` instance,
* `function.injective.ordered_comm_semiring`, with the `to_ordered_comm_semiring` instance,
* `function.injective.ordered_ring`.

For the last one (`ordered_ring`), since there are opposites, there is no way around having to play with opposites, so I added some assumptions.  This means that, in order to get the instance, we also need to assume something of the `subsemiring`: for instance, the natural numbers are a subsemiring of the integers, the integers are an ordered_ring, but the naturals still have no opposites!

I will leave this for later, if this is ok.

Reply:
For the ring cases, the instances should attach to subring, not subsemiring